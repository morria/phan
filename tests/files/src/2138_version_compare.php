<?php  declare(strict_types=1);

$cond1 = version_compare(); // PhanParamTooFewInternal
$cond2 = version_compare('0.1.1', '0.1.2-rc', '>=');
assert_int(version_compare('0.1.1', '0.1.2-rc')); // ok
assert_int(version_compare('0.1.1', '0.1.2-rc', '>='));
assert_bool(version_compare('0.1.1', '0.1.2-rc'));
assert_bool(version_compare('0.1.1', '0.1.2-rc', '>=')); // ok

function assert_int(int $v) {}
function assert_bool(bool $v) {}
