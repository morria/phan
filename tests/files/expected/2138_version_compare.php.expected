%s:3 PhanParamTooFewInternal Call with 0 arg(s) to \version_compare() which requires 2 arg(s)
%s:6 PhanTypeMismatchArgument Argument 1 (v) is bool but \assert_int() takes int defined at %s:10
%s:7 PhanTypeMismatchArgument Argument 1 (v) is int but \assert_bool() takes bool defined at %s:11
